aten/src/ATen/native/cpu/MultinomialKernel.cpp

```diff
@@ -94,25 +92,19 @@ multinomial_with_replacement_apply(
       ie cum_dist[row][slot-1] < uniform_prob < cum_distr[row][slot] */
       int left_pointer = 0;
       int right_pointer = n_categories;
-      // NOLINTNEXTLINE(cppcoreguidelines-init-variables)
-      int mid_pointer;
-      // NOLINTNEXTLINE(cppcoreguidelines-pro-type-member-init)
-      scalar_t cum_prob;
-      // NOLINTNEXTLINE(cppcoreguidelines-init-variables)
-      int sample_idx;
       /* Make sure the last cumulative distribution bucket sums to 1 */
       cum_dist_ptr[(n_categories - 1) * cum_dist_stride_0] = 1;
 
       while (right_pointer - left_pointer > 0) {
-        mid_pointer = left_pointer + (right_pointer - left_pointer) / 2;
-        cum_prob = cum_dist_ptr[mid_pointer * cum_dist_stride_0];
+        auto mid_pointer = left_pointer + (right_pointer - left_pointer) / 2;
```

Comment:
I'm not sure about these auto. The exact width of the integer type is important for these. Let's keep explicit types.

Reply:
This is used in binary search, should be deduced to the pointer type.